himbaechel/uarch/gowin/gowin.cc

```diff
@@ -1059,6 +1059,28 @@ bool GowinImpl::slice_valid(int x, int y, int z) const
                 return false;
             }
         }
+        // Check whether the current architecture allows 6 and 7 DFFs
+        if (z > 3 && ctx->getBelByLocation(Loc(x, y, 6 * 2 + 1)) != BelId()) {
```

Comment:
I feel like there has to be a better way of doing this, personally. In the gatemate arch we have an "extra data" type associated with logic, and you would set a flag that says "this arch has 6/7 DFFs" or so. But, if it works, why not I suppose.

Reply:
Well, in principle, it can be changed :)